backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -551,4 +564,30 @@ private static void checkSystemEndpoint(EndpointType endpointType) {
     private boolean isEndpointTypeAllowed(EndpointType endpointType) {
         return !featureFlipper.isEmailsOnlyMode() || endpointType.isSystemEndpointType;
     }
+
+    /**
+     * Removes the secrets from the endpoint's properties when returning them
+     * to the client.
+     * @param endpoint the endpoint to redact the secrets from.
+     */
+    @Deprecated(forRemoval = true)
+    protected void redactSecretsForEndpoint(final Endpoint endpoint) {
+        if (endpoint.getProperties() instanceof SourcesSecretable sourcesSecretable) {
+            final BasicAuthentication basicAuthentication = sourcesSecretable.getBasicAuthentication();
+            if (basicAuthentication != null) {
+                basicAuthentication.setPassword("[REDACTED CREDENTIAL]");
+                basicAuthentication.setUsername("[REDACTED CREDENTIAL]");
+            }
+
+            final String bearerToken = sourcesSecretable.getBearerAuthentication();
+            if (bearerToken != null) {
+                sourcesSecretable.setBearerAuthentication("[REDACTED CREDENTIAL]");
+            }
+
+            final String secretToken = sourcesSecretable.getSecretToken();
+            if (secretToken != null) {
+                sourcesSecretable.setSecretToken("[REDACTED CREDENTIAL]");
```

Comment:
replacement string should be a constant.

Reply:
Done!